src/commands/format.rs

```diff
@@ -188,7 +196,7 @@ pub fn format(args: FormatArgs) -> Result<()> {
 
     let mut diagnostics = 0;
     if args.path.to_str() != Some("-") && args.path.is_dir() {
-        if !args.overwrite && !args.check {
+        if !args.mode.overwrite && !args.mode.check {
             bail!("formatting a directory requires the `--overwrite` or `--check` option");
         }
```

Comment:
Now that these are required inputs, do we still need this check?

Reply:
Agree with Andrew. This check can be removed.

Reply:
```suggestion
```